src/50_Samples_%26_Templates/Product.html

```diff
@@ -335,69 +335,71 @@ <h2 class="m1">Apple</h2>
     corresponding to the selected colour. We realized the color selection by using `amp-selector`. Find more the doc and info in The Product Page section below.
 
     The green colour is the default color for the gallery when the page loads.
-    Clicking on a preview image will reveal the image in the gallery via the tap action `on="tap:product-carousel-preview.goToSlide(index=0)"`.
+    Clicking on a preview image will reveal the image in the gallery via the tap action `on="tap:AMP.setState({selectedSlide: 0})"`.
 
     See below for more informations on how we use`amp-bind`. -->
   <div class="product-gallery">
-    <ul class="show" [class]="(!selectedColor || selectedColor == 'green') ? 'show' : 'hide'">
+    <ul class="show" [class]="(!product.selectedColor || product.selectedColor == 'green') ? 'show' : 'hide'">
```

Comment:
yes, but the sample depends on the value of variables when the page loads. For example this code:
`<p class="price-description">Price:
        <span [text]="product[selectedColor || 'green'].sizes[selectedSize || 'S']">$1.99</span>
    </p>`
note that it will launch an error when the page loads if we don't check for null.